packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -45,6 +45,39 @@ class PythonModule {
     'Install via \'pip install $name\' or \'sudo easy_install $name\'.';
 }
 
+class IMobileDevice {
+  const IMobileDevice();
+
+  static IMobileDevice get instance => context.putIfAbsent(IMobileDevice, () => const IMobileDevice());
+
+  bool get isInstalled => exitsHappy(<String>['idevice_id', '-h']);
+
+  /// Returns true if libimobiledevice is installed and working as expected.
+  ///
+  /// Older releases of libimobiledevice fail to work with iOS 10.3 and above.
+  Future<bool> get isWorking async {
+    if (!isInstalled)
+      return false;
+
+    // If a device is attached, verify that we can get its name.
+    final ProcessResult result = (await runAsync(<String>['idevice_id', '-l'])).processResult;
+    if (result.exitCode == 0 && result.stdout.isNotEmpty && !await exitsHappyAsync(<String>['idevicename']))
+      return false;
+    return true;
+  }
+
+  List<String> getAttachedDeviceIDs() {
```

Comment:
Painful -- it cascades down to a whole bunch of API changes to shared iOS and Android interfaces :(